File: src/com/eagle/programmar/COBOL/Terminals/COBOL_Punctuation.java
// Copyright Eagle Legacy Modernization, 2010-date
// Original author: Steven A. O'Hara, Dec 6, 2010

package com.eagle.programmar.COBOL.Terminals;

import com.eagle.core.EagleSyntax;
import com.eagle.parsers.EagleFileReader;
import com.eagle.parsers.EagleLineReader;
import com.eagle.tokens.TerminalPunctuationToken;

public class COBOL_Punctuation extends TerminalPunctuationToken
{
	// Need default constructor for reading from the XML file
	public COBOL_Punctuation()
	{
		this('\0');
	}

	public COBOL_Punctuation(char punct)
	{
		super(punct);
	}

	public COBOL_Punctuation(String punct)
	{
		super(punct);
	}

	@Override
	public boolean parse(EagleFileReader lines)
	{
		if (findStart(lines) == FOUND.EOF) return false;
		if (_punct1 == '*')
		{
			// Make sure a '*' is not a comment
			EagleLineReader rec = lines.get(_currentLine);
			if (rec.charAt(_currentChar) == '*')
			{
				EagleSyntax syntax = this.getSyntax();
				if (_currentChar == syntax._commentColumn) return false;
			}
		}
		return super.parse(lines);
	}
}

File: csrc/com/eagle/programmar/COBOL/Terminals/COBOL_Punctuation.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// Purchase a Premium Edition license at:
// https://www.tangiblesoftwaresolutions.com/order/order-java-to-csharp.html
// ====================================================================================================

// Copyright Eagle Legacy Modernization, 2010-date
// Original author: Steven A. O'Hara, Dec 6, 2010

namespace com.eagle.programmar.COBOL.Terminals
{
	using EagleSyntax = com.eagle.core.EagleSyntax;
	using EagleFileReader = com.eagle.parsers.EagleFileReader;
	using EagleLineReader = com.eagle.parsers.EagleLineReader;
	using TerminalPunctuationToken = com.eagle.tokens.terminals.TerminalPunctuationToken;

	public class COBOL_Punctuation : TerminalPunctuationToken
	{
		// Need default constructor for reading from the XML file
		public COBOL_Punctuation() : this('\0')
		{
		}

		public COBOL_Punctuation(char punct) : base(punct)
		{
		}

		public COBOL_Punctuation(string punct) : base(punct)
		{
		}

		public override bool parse(EagleFileReader lines)
		{
			if (findStart(lines) == FOUND.EOF)
			{
				return false;
			}
			if (_punct1 == '*')
			{
				// Make sure a '*' is not a comment
				EagleLineReader rec = lines.get(_currentLine);
				if (rec.charAt(_currentChar) == '*')
				{
					EagleSyntax syntax = this.getSyntax();
					if (_currentChar == syntax._commentColumn)
					{
						return false;
					}
				}
			}
			return base.parse(lines);
		}
	}

}
